donotsubmit/action.yaml

```diff
@@ -0,0 +1,40 @@
+# Copyright 2022 Chainguard, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+name: 'Do Not Submit'
+description: |
+  This action flags occurrences of the string "DO NOT SUBMIT"
+  throughout the source tree.  This enables developers to flag
+  pieces of code they intend to revisit before submitting with
+  this string and have CI block the change from merging.
+
+inputs: {}
+
+runs:
+  using: "composite"
+  steps:
+  - name: Do Not Submit
+    shell: bash
+    run: |
+      set -e
+      cd "${GITHUB_WORKSPACE}" || exit 1
+
+      TEMP_PATH="$(mktemp -d)"
+      PATH="${TEMP_PATH}:$PATH"
+
+      echo '::group::🐶 Installing reviewdog ... https://github.com/reviewdog/reviewdog'
+      curl -sfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh | sh -s -- -b "${TEMP_PATH}" 2>&1
```

Comment:
preference between curl install or reviewdog action?

https://github.com/reviewdog/reviewdog#github-actions

Reply:
Hmm, we can revisit this.  It's been forever since we first wrote this, and this seems much nicer:
```yaml
      - uses: reviewdog/action-setup@v1
        with:
          reviewdog_version: latest # Optional. [latest,nightly,v.X.Y.Z]
```

Reply:
Done